assets/js/services/coinify.service.js

```diff
@@ -0,0 +1,247 @@
+    },
+    get balanceAboveMax () {
+      return service.sellMax && service.sellMax > service.limits.blockchain.outRemaining['BTC'];
+    },
+    get buyLimitRemaining () {
+      let { limits } = service;
+      let { defaultCurrency } = service.exchange.profile;
+      return Math.max(limits.bank.inRemaining[defaultCurrency], limits.card.inRemaining[defaultCurrency]);
+    },
+    get userCanBuy () {
+      return service.userCanTrade;
+    },
+    get userCanSell () {
+      return service.userCanTrade && service.balanceAboveMin;
+    },
+    get disabledUntil () {
+      return service.exchange.profile && Math.ceil((service.exchange.profile.canTradeAfter - Date.now()) / ONE_DAY_MS);
+    },
+    get buyReason () {
+      let reason;
+      let { profile, user } = service.exchange;
+
+      if (!user) reason = 'user_needs_account';
+      else if (!profile.canTrade) reason = profile.cannotTradeReason;
+      else reason = 'has_remaining_buy_limit';
+
+      return reason;
+    },
+    get sellReason () {
+      let reason;
+      let { profile, user } = service.exchange;
+
+      if (user && !profile.canTrade) reason = profile.cannotTradeReason;
+      else if (service.balanceAboveMin) reason = 'can_sell_remaining_balance';
+      else if (!service.balanceAboveMin) reason = 'not_enough_funds_to_sell';
+      else if (service.balanceAboveMax) reason = 'can_sell_max';
+      else reason = 'has_remaining_sell_limit';
+
+      return reason;
+    },
+    get buyLaunchOptions () {
+      let reason = service.buyReason;
+      let { user, profile } = service.exchange;
+
+      if (reason === 'has_remaining_buy_limit' && user && +profile.level.name < 2) return { 'KYC': service.openPendingKYC };
+      else if (reason === 'awaiting_first_trade_completion' && service.getPendingTrade()) return { 'FINISH': service.openPendingTrade, 'CANCEL': service.cancelTrade };
+      else if (reason === 'awaiting_first_trade_completion' && service.getProcessingTrade()) return { 'CHECK_STATUS': service.openProcessingTrade };
+      else if (reason === 'after_first_trade') return { 'WHY': service.openTradingDisabledHelper };
+    },
+    get sellLaunchOptions () {
+      let reason = service.sellReason;
+      let { user, profile } = service.exchange;
+
+      if (reason === 'can_sell_max' && user && profile.level && +profile.level.name < 2) return { 'KYC': service.openPendingKYC };
+      else if (reason === 'not_enough_funds_to_sell') return { 'REQUEST': modals.openRequest, 'BUY': service.goToBuy };
+      else if (reason === 'awaiting_first_trade_completion' && service.getPendingTrade()) return { 'FINISH': service.openPendingTrade, 'CANCEL': service.cancelTrade };
+      else if (reason === 'awaiting_first_trade_completion' && service.getProcessingTrade()) return { 'CHECK_STATUS': service.openProcessingTrade };
+      else if (reason === 'after_first_trade') return { 'WHY': service.openTradingDisabledHelper };
```

Comment:
Duplication between `buyLaunchOptions` and `sellLaunchOptions`. It's not necessarily bad, but maybe this can be improved.

Reply:
Any ideas how this can be improved?